portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -57,15 +57,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    // store each comment in a comment object
+    long numComments = 0;
+    boolean firstComment = true;
+    long commentsAdded = 0;
+
+    // stores each comment in a comment object
+    // for the last commment submitted, stores the number of comments to print
+    // only creates a comment object for the given number of comments to print
     ArrayList<Comment> comments = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
-      long timestamp = (long) entity.getProperty("timestamp");
-      String userComment = (String) entity.getProperty("stringValue");
-      
-      Comment comment = new Comment(id, userComment, timestamp);
-      comments.add(comment);
+      if(firstComment){
+          numComments = (long) entity.getProperty("numComments");
+      }
+      if(commentsAdded < numComments){
+        long id = entity.getKey().getId();
+        long timestamp = (long) entity.getProperty("timestamp");
+        String userComment = (String) entity.getProperty("stringValue");
+        Comment comment = new Comment(id, userComment, timestamp);
+        comments.add(comment);
+        commentsAdded++;
+      }
+      firstComment = false;
```

Comment:
this should likely go in the if(firstComment) block. It'll have the same functionality but will have the benefit of being clearer to a reader of the code what's happening. We don't need this value to be true in the rest of the usage so it's best to set it to false ASAP.

Reply:
That makes sense, I switched it.